src/composables/useLeaflet.ts

```diff
@@ -0,0 +1,138 @@
+import { LMap } from "@vue-leaflet/vue-leaflet";
+import { Feature, Geometry } from "geojson";
+import {
+    LatLngBounds,
+    Map,
+    geoJSON,
+    GeoJSON,
+    Polyline,
+    Layer,
+    polyline,
+    StyleFunction,
+    TooltipOptions,
+    LeafletEventHandlerFnMap,
+    GeoJSONOptions
+} from "leaflet";
+import { storeToRefs } from "pinia";
+import { useAppStore } from "../stores/appStore";
+import { minZoom } from "../utils";
+
+type FeatureProperties = { GID_0: string; GID_1: string; NAME_1: string };
+type TooltipOptionAndContent = { content: string; options?: TooltipOptions };
+// laeflet geojson options do not include smoothFactor, probably outdated
+type GeojsonOptions = GeoJSONOptions<FeatureProperties, Geometry> & { smoothFactor: number };
+
+export const useLeaflet = (
+    style: StyleFunction,
+    getTooltip: (f: Feature) => TooltipOptionAndContent,
+    layerEvents: (f: Feature) => LeafletEventHandlerFnMap
+) => {
+    // external refs: map related
+    const map = shallowRef<typeof LMap | null>(null);
+    const getLeafletMap = () => map.value!.leafletObject as Map | undefined;
+
+    // external refs: bounds related
+    const bounds = ref<LatLngBounds | null>(null);
+    const lockBounds = ref<boolean>(false);
```

Comment:
I think a comment on what this is and how it will be used would be good.

Reply:
done